examples/src/createReactAgent/accessThreadLevelStateInTools.ts

```diff
@@ -0,0 +1,335 @@
+/**
+ * Access Thread Level Context in Tools
+ *
+ * This allows tools to access the current conversation history and thread-specific state during execution, enabling tools to make informed decisions based on the ongoing dialogue.
+ *
+ * Why this is important:
+ * - Conversation-Aware Tools:
+ *   Tools can reference previous interactions and maintain consistency with the ongoing conversation
+ * - Context-Dependent Actions:
+ *   Tool behavior can vary based on what has been discussed or accomplished earlier in the thread
+ * - Intelligent Workflow Management:
+ *   Enables tools to coordinate with each other and avoid redundant actions
+ *
+ * Example Scenario:
```

Comment:
maybe better example: using a sub agent as a tool. you want to pass all message to sub agent which is inside tool